### PR TITLE
refactor: [M3-8815] – Migrate `TextField` component to `ui` package

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Migrate `TextField` component to `ui` package.

## Changes  🔄
- Migrate `TextField` component to `ui` package
- Migrate `convertToKebabCase` utility to `ui` package (to avoid having `manager` be a dependency of `ui`)
- Migrate `Plus` icon to `ui` package

## Target release date 🗓️
12/10/24

### Verification steps
- All tests should pass
- There should be no regressions observed in Cloud
- `TextField` story in Storybook should be in good shape

## As an Author, I have considered 🤔
- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [X] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [X] All unit tests are passing
- [X] TypeScript compilation succeeded without errors
- [X] Code passes all linting rules

### CODE REVIEW

**Comment by mjac0bs** [comment]
@dwiley-akamai This looks like it will need a rebase on `develop` to make CI happy; Alban fixed the issue [here](https://github.com/linode/manager/pull/11289/commits/ff321d594a8ac859f8abdaea6890fe499634bd14).

File: packages/ui/src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -5,15 +5,14 @@ import {
   InputAdornment,
   InputLabel,
   TooltipIcon,
+  convertToKebabCase,
 } from '@linode/ui';
```
Comment (harsh-akamai) [inline packages/ui/src/components/TextField/TextField.tsx:9]: Can we avoid circular dependencies inside the ui package?

File: packages/ui/src/components/TextField/TextField.tsx [hunk 2]
```diff
@@ -5,15 +5,14 @@ import {
...
 import KeyboardArrowDown from '@mui/icons-material/KeyboardArrowDown';
 import { useTheme } from '@mui/material/styles';
 import { default as _TextField } from '@mui/material/TextField';
 import { clamp } from 'ramda';
 import * as React from 'react';
 
-import { convertToKebabCase } from 'src/utilities/convertToKebobCase';
-
 import type { BoxProps, TooltipProps } from '@linode/ui';
```
Comment (harsh-akamai) [inline packages/ui/src/components/TextField/TextField.tsx:16]: Same here

**Review by harsh-akamai — COMMENTED** [review]
There are multiple imports of the ui package inside the `TextField` folder. Replacing it with relative imports will avoid circular dependency.

File: packages/ui/src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
-  InputLabel,
-  TooltipIcon,
-} from '@linode/ui';
+import { Box } from '../Box';
+import { CircleProgress } from '../CircleProgress';
+import { FormHelperText } from '../FormHelperText';
+import { InputAdornment } from '../InputAdornment';
+import { InputLabel } from '../InputLabel';
+import { TooltipIcon } from '../TooltipIcon';
+import { convertToKebabCase } from '../../utilities';
```
Comment (coliu-akamai) [inline packages/ui/src/components/TextField/TextField.tsx:7]: not a big deal but do we have eslint set up for import order in the `ui` package? I feel like I normally see `'../imports'` go after `'@imports'`, but I'm not seeing any of those eslint squiggles in vscode (they appear in the `manager` package though)

**Review by coliu-akamai — APPROVED** [review]
✅ storybook
✅ general check of CM
✅ code review

Thanks Dajahi! 🎉

**Review by mjac0bs — COMMENTED** [review]
Noticed no regressions to TextField in the UI. I'm just seeing Storybook and unit test issues locally that I don't fully understand. If others more familiar with the migration have no concerns with this PR, then feel free to merge. 👍🏼 

Storybook's error when trying to view TextField locally:
image
- Removing the node_modules cache doesn't fix the issue.
- yarn clean && yarn up doesn't fix the issue

When running unit tests locally (root level or in /manager package), I'm seeing failures but they appear to be in Database tests, and `yarn test` passes within the `ui` package. CI's unit tests passed as well, so I don't *think* it's due to this PR.

**Review by mjac0bs — APPROVED** [review]
Okay, disregard the earlier comment. Some combination of cache clearing, cleaning, browser history clearing, and restarting tests and Storybook got things running locally for me as expected. 🚢 

Screenshot 2024-11-20 at 12 13 47 PM
image

File: packages/ui/src/components/TextField/TextField.tsx (hunk 3 above)
Comment (dwiley-akamai) [reply packages/ui/src/components/TextField/TextField.tsx:7]: I think https://github.com/linode/manager/pull/11258 might be adding this

**Review by harsh-akamai — APPROVED** [review]

**Pull request merged**